tests/e2e/specs/shopper/mini-cart.test.js

```diff
@@ -7,12 +12,33 @@ const block = {
 	name: 'Mini Cart Block',
 };
 
+const options = getDefaultOptions();
+
+const clickMiniCartButton = async () => {
+	await page.hover( '.wc-block-mini-cart__button' );
+
+	await expect( page ).not.toMatchElement(
+		'.wc-block-mini-cart__drawer.is-loading'
+	);
+
+	await page.click( '.wc-block-mini-cart__button' );
+};
```

Comment:
Can we rename this to represent outcome more than user action `openMiniCartDrawer` so that it's easier to understand the state in which the test is?

Also thinking if we should have `expect` in here or if it should be part of a separate test ie. `Mini Cart drawer is initially closed` and then we could use `page.waitForXPath` option `hidden: true` to wait for loading class to go away and have this as a pure util.

then we could return `page.waitForXPath` from this action helper and use its return value in expect to ensure drawer is open. 

This way we wouldn't have to test for drawer being closed and open repeatedly and just ensure it in initial tests focused on these two states. 

What do you think?

Reply:
> Can we rename this to represent outcome more than user direct user action ie. `openMiniCartDrawer` so that it's easier to understand the state in which the test is?

The drawer can be opened by clicking the mini cart button or adding a product to the cart, so I don't think `openMiniCartDrawer` is fully precise here. What do you think?

> Also thinking if we should have `expect` in here or if it should be part of a separate test ie. `Mini Cart drawer is initially closed` and then we could use `page.waitForXPath` option `hidden: true` to wait for loading class to go away and have this as a pure util.

I removed `expect` from `clickMiniCartButton`.

> then we could return `page.waitForXPath` from this action helper and use its return value in expect to ensure drawer is open.
> 
> This way we wouldn't have to test for drawer being closed and open repeatedly and just ensure it in initial tests focused on these two states.

I do not fully understand your idea, could you please explain it in more detail?